api/notifications_and_subscriptions.rb

```diff
@@ -28,3 +28,26 @@
 delete "#{APIPREFIX}/users/:user_id/subscriptions" do |user_id|
   user.unsubscribe(source).to_hash.to_json
 end
+
+get "#{APIPREFIX}/subscriptions/:thread_id" do |thread_id|
+  page = (params['page'] || DEFAULT_PAGE).to_i
+  per_page = (params['per_page'] || DEFAULT_PER_PAGE).to_i
+
+  # Build a query hash based on the query parameters
+  query = {}
+  query[:subscriber_id] = params[:subscriber_id] if params[:subscriber_id]
+  query[:source_id] = thread_id
+  query[:source_type] = params[:source_type] if params[:source_type]
+
+  subscriptions = Subscription.where(query).paginate(:page => page, :per_page => per_page)
+  subscriptions_count = subscriptions.total_entries
+
+  content_type :json
+
+  {
+    collection: subscriptions.map(&:to_hash),
```

Comment:
do we need to return the whole `collection`, or only the ids?

Reply:
subscription object has only 4 attributes, so I guess we can return all without taking any performance hit.